src/types/adapters/IntegrationAdapter.ts

```diff
@@ -69,62 +70,71 @@ const toSecretToken = (
+  properties?: Schemas.WebhookPropertiesDTO
 ): IntegrationAnsible => ({
   ...integrationBase,
   url: properties?.url ?? '',
-  sslVerificationEnabled: !properties?.disable_ssl_verification ?? false,
-  secretToken: toSecretToken(properties?.secret_token),
+  sslVerificationEnabled: !properties?.disableSslVerification ?? false,
+  secretToken: toSecretToken(properties?.secretToken),
   method: properties?.method ?? Schemas.HttpType.Enum.POST,
 });
 
 const toIntegrationCamel = (
   integrationBase: IntegrationBase<CamelIntegrationType>,
-  properties?: Schemas.CamelProperties
+  properties?: Schemas.CamelPropertiesDTO
 ): IntegrationCamel => ({
   ...integrationBase,
   url: properties?.url ?? '',
-  sslVerificationEnabled: !properties?.disable_ssl_verification ?? false,
-  secretToken: toSecretToken(properties?.secret_token),
+  sslVerificationEnabled: !properties?.disableSslVerification ?? false,
+  secretToken: toSecretToken(properties?.secretToken),
   basicAuth:
-    properties?.basic_authentication === null
+    properties?.basicAuthentication === null
       ? undefined
       : {
-          user: notNull(properties?.basic_authentication?.username, ''),
-          pass: notNull(properties?.basic_authentication?.password, ''),
+          user: notNull(properties?.basicAuthentication?.username, ''),
+          pass: notNull(properties?.basicAuthentication?.password, ''),
         },
   extras: notNull(properties?.extras),
 });
 
 const toIntegrationEmail = (
   integrationBase: IntegrationBase<IntegrationType.EMAIL_SUBSCRIPTION>,
-  properties: Schemas.EmailSubscriptionProperties
+  properties: Schemas.SystemSubscriptionPropertiesDTO
 ): IntegrationEmailSubscription => ({
   ...integrationBase,
-  ignorePreferences: properties.ignore_preferences,
-  groupId: properties.group_id === null ? undefined : properties.group_id,
-  onlyAdmin: properties.only_admins,
+  ignorePreferences: properties.ignorePreferences,
+  groupId: properties.groupId === null ? undefined : properties.groupId,
+  onlyAdmin: properties.onlyAdmins === null ? undefined : properties.onlyAdmins,
 });
 
 const toIntegrationDrawer = (
   integrationBase: IntegrationBase<IntegrationType.DRAWER>,
-  properties: Schemas.DrawerProperties
+  properties: Schemas.SystemSubscriptionPropertiesDTO
 ): IntegrationDrawer => ({
   ...integrationBase,
-  ignorePreferences: properties.ignore_preferences,
-  groupId: properties.group_id === null ? undefined : properties.group_id,
-  onlyAdmin: properties.only_admins,
+  ignorePreferences: properties.ignorePreferences,
+  groupId: properties.groupId === null ? undefined : properties.groupId,
+  onlyAdmin: properties.onlyAdmins,
+});
+
+const toIntegrationPagerDuty = (
+  integrationBase: IntegrationBase<IntegrationType.PAGERDUTY>,
+  properties: Schemas.PagerDutyPropertiesDTO
+): IntegrationPagerduty => ({
+  ...integrationBase,
+  secretToken: properties.secretToken,
+  severity: properties.severity,
 });
```

Comment:
Right now the server responses with secret_token which is deviation from generated code, but that might be because of stale client generated. We have a ticket to aligh with the new spec, but in the meantime we could map the value a bit differently
```suggestion
const toIntegrationPagerDuty = (
  integrationBase: IntegrationBase<IntegrationType.PAGERDUTY>,
  properties: Schemas.PagerDutyPropertiesDTO & { secret_token?: string }
): IntegrationPagerduty => ({
  ...integrationBase,
  secretToken: properties.secret_token || properties.secretToken,
  severity: properties.severity,
});
```